docs/directory-structure.md

```diff
@@ -1,241 +1,213 @@
+repo, meant to replace the somewhat ad-hoc layout of early 2021, in order
+to:
+
+  - Allow generalization to more languages than Plains Cree, and
+
+  - Address some other issues that have arisen with the source code layout.
+
+## The proposal
+
+### High-level decisions / assumptions
+
+  - In the file system, dictionary applications such as itwêwina are named
+    `sssttt`, where `sss` and `ttt` are each three-letter [ISO 639-3
+    language codes][639-3] for the dictionary [source language and target
+    language](glossary.md), respectively. Example: `crkeng`.
+
+    The use of a technical abbreviation means that nobody will ever be
+    blocked from starting a dictionary for a new language pair on on
+    naming/branding questions.
+
+  - There will be at most one intelligent dictionary application for each
+    language pair. The code already supports multiple dictionary sources
+    within one application, such as both the *Cree: Words* and Maskwacîs
+    dictionaries for Plains Cree.
+
+  - Each site will be an independently deployed django project, and have an
+    independent database.
+
+    There are advantages and disadvantages to having a single django
+    process serving multiple sites, or having multiple sites share a single
+    database; however, this proposed compartmentalization should allow
+    reduced risk when experimenting with new languages.
+
+  - For now, all the code for all the dictionary applications will reside
+    in this git repo, and not be split into separate git repos.
+
+    That will make it much easier to move code around during development,
+    and to run tests across all dictionary applications when making
+    changes.
+
+    Some day, we hope, when mature and stable, morphodict could go on PyPI
+    as its own framework package with instructions on how to set up new
+    languages without being in this git repo. We are very far from that
+    point right now, but it’s something to keep in mind as a long-term
+    goal.
+
+[639-3]: https://iso639-3.sil.org/code_tables/639/data
+
+### Proposed layout
+
+    $repo
+    ├── .git/
+    ├── package.json                # dependencies for bundlers, JS/CSS frameworks
+    ├── Pipfile
+    ├── arpeng-manage.py            # manage.py scripts are at the top-level for easy access
+    ├── crkeng-manage.py
+    ├── crkfra-manage.py
+    ├── cwdeng-manage.py
+    ├── srseng-manage.py
+    └── src/
+        ├── CreeDictionary/         # existing code, eventually goes away
+        │   ├── __init__.py
+        │   ├── API/                # this name goes away :(
+        │   └── CreeDictionary/     # this goes away too
+        │       ├── __init__.py
+        │       ├── models.py
+        │       ├── views.py
+        │       └── …
+        │
+        ├── morphodict/             # python package for language-independent code
+        │   ├── __init__.py
+        │   ├── cvd/
+        │   ├── lexicon/            # django app with primary database tables
```

Comment:
Will all the language data live in a single database, or will we run separate database instances for each language?

I like having a single database, because it forces us to structure language data the same way across languages.

Reply:
We will have separate databases for each language pair so that time-consuming/risky database operations for new languages we’re experimenting with won’t affect production sites like itwêwina.

But, the code to access the database will be shared, and thanks to Django, each database will have exactly the same *schema* for the same tables.